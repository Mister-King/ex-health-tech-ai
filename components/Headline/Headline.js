/**
 * ButtonContainer component
 */

import React from 'react';
import { Text } from 'react-native';

import styles from './_headline.scss';

const Headline = props => {
    let classNames = [styles.headline];
    props.type  && classNames.push(styles[`headline--${props.type}`]);
    props.style && classNames.push(props.style);

    return (
        <Text className={classNames}>
            {props.children}
        </Text>
    );
};

export default Headline;
