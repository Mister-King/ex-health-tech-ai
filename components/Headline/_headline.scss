/**
 * Headline
 */

@import '../variables';

.headline {
	font-size: 32px;
	line-height: 40px;
	font-weight: 700;
	margin: 40px 0 20px;
	text-align: center;
	color: map-get($colours, 'copy');

	&--secondary {
		font-size: 24px;
		line-height: 32px;
		margin-bottom: 20px;
	}

	&--tertiary {
		font-size: 16px;
		line-height: 20px;
		margin-bottom: 10px;
	}
}
