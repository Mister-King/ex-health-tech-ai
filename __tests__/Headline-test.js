/* eslint no-undef: 0 */
/**
 * @format
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Headline from '../components/Headline/Headline';

// Note: test renderer must be required after react-native.

test('renders default headline', () => {
    const tree = renderer.create(
        <Headline>Test Headline</Headline>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

test('renders secondary headline', () => {
    const tree = renderer.create(
        <Headline type="secondary">Test Secondary Headline</Headline>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

test('renders custom styles headline', () => {
    const tree = renderer.create(
        <Headline className="test-styles">Test Custom Styles Headline</Headline>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
