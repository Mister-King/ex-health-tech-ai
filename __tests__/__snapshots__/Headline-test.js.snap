// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders custom styles headline 1`] = `
<Text
  style={
    Array [
      "headline",
    ]
  }
>
  Test Custom Styles Headline
</Text>
`;

exports[`renders default headline 1`] = `
<Text
  style={
    Array [
      "headline",
    ]
  }
>
  Test Headline
</Text>
`;

exports[`renders secondary headline 1`] = `
<Text
  style={
    Array [
      "headline",
      "headline--secondary",
    ]
  }
>
  Test Secondary Headline
</Text>
`;
